packages/flutter/lib/src/widgets/implicit_animations.dart

```diff
@@ -1173,3 +1173,107 @@ class _AnimatedPhysicalModelState extends AnimatedWidgetBaseState<AnimatedPhysic
     );
   }
 }
+
+/// Animated version of [PhysicalShape].
+///
+/// The [elevation] is animated.
+///
+/// The [color] is animated if the [animateColor] property is set; otherwise,
+/// the color changes immediately at the start of the animation for the other
+/// two properties. This allows the color to be animated independently (e.g.
+/// because it is being driven by an [AnimatedTheme]).
+///
+/// The [clipper] is not animated.
```

Comment:
This is the only serious problem here. An AnimatedPhysicalShape whose shape can't be animated doesn't seem right.

Reply:
We are planning to use the shape for drawing a border as well, we should animate the shape at an ancestor that is aware both of the PhysicalShape and of the border.

They way things are set up right now - AnimatedPhysicalShape/PhysicalShape is unaware of the border.

A few options we brought up in our discussion on this:
1. Make PhysicalShape draw the border, or add a new widget which is a PhysicalShape with border.
2. Make Material animate the shape, and hand it over to AnimatedPhysicalShape and to the widget that's drawing the border.
3. Make Material animate shape, color, shadowColor, and elevation, and use PhysicalShape directly.
4. Have a material widget which is not animated, and another implicitly animated material widget the animates all these properties.

Current CL was done under the assumption that <i>#</i>2 is our plan, but we are now discussing this again.